src/Framework/Framework/Compilation/Binding/TypeRegistry.cs

```diff
@@ -34,8 +35,13 @@ public TypeRegistry(CompiledAssemblyCache compiledAssemblyCache, ImmutableDictio
             {
                 return expr;
             }
-            expr = resolvers.Select(r => r(name)).FirstOrDefault(e => e != null);
-            if (expr != null) return expr;
+            var matchedExpressions = resolvers.Select(r => r(name)).Where(e => e != null).Distinct(ExpressionTypeComparer.Instance).ToList();
```

Comment:
<span title="unimportant suggestion">:information_source:</span> We have the `WhereNotNull` extension method, which will allow you to make the ExpressionTypeComparer non-nullable